src/test/java/com/google/api/codegen/gapic/testdata/ruby/ruby_library.baseline

```diff
@@ -4970,311 +4970,882 @@ Gem::Specification.new do |gem|
+
+  operation = client.get_big_book(formatted_name) do |op|
+    raise op.results.message if op.error?
+    response = op.results
+    # Testing iterating over map fields when both key and value are specified.
+    response.map_list_value_value.each do |my_key, my_value|
+      puts "key: #{my_key}, value: #{my_value}"
+    end
 
-  metadata = op.metadata
-  # Process the metadata.
+    # Testing iterating over map fields when only key is specified.
+    response.map_list_value_value.keys.each do |another_key|
+      puts "key: #{another_key}"
+    end
+
+    # Testing iterating over map fields when only value is specified.
+    response.map_list_value_value.values.each do |another_value|
+      puts "value: #{another_value}"
+    end
+
+    puts "name: #{response.name}"
+    puts "author: #{response.author}"
+
+    metadata = op.metadata
+    # Process the metadata.
+  end
+
+  operation.wait_until_done!
+
+  # [END hopper_core]
 end
+# [END hopper]
+
+
+require "optparse"
+
+if $0 == __FILE__
+
+  shelf = "Novel"
+
+  ARGV.options do |opts|
+    opts.on("--shelf=val") { |val| shelf = val }
+    opts.parse!
+  end
+
 
-operation.wait_until_done!
-# FIXME: response handling
-# FIXME: main function
+  sample_get_big_book(shelf)
+end
 ============== file: samples/get_big_book_long_running_request_async_wap2.rb ==============
-//FIXME: license section
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
 # DO NOT EDIT! This is a generated sample ("LongRunningRequestAsync",  "wap2")
-# FIXME: import section
-# FIXME: sample function declaration
-# FIXME: initialize client
 
-formatted_name = Library::V1::LibraryServiceClient.book_path(shelf, big_book_name)
+require "library"
+
+# [START hopper]
+
+ # Testing resource name overlap
+ #
+ # @param shelf {String} Test word wrapping for long lines. This is a long comment. The name of the
+ # shelf to retrieve the big book from.
+ # @param big_book_name {String} The name of the book.
+def sample_get_big_book(shelf, big_book_name)
+  # [START hopper_core]
+  # Instantiate a client
+  client = Library.new(version: :v1)
+
+  formatted_name = Library::V1::LibraryServiceClient.book_path(shelf, big_book_name)
 
-operation = library_service_client.get_big_book(formatted_name) do |op|
-  raise op.results.message if op.error?
-  op_results = op.results
-  # Process the results.
+  operation = client.get_big_book(formatted_name) do |op|
```

Comment:
Our Ruby API code samples use the block syntax but I don't think it's as clear to our users as:
  - a. kick-off operation
  - b. block til complete
  - c. get the response (or error)

Can we rock this syntax?

```ruby
#... <code to construct request object> ...

# Make the long-running operation request
operation = speech.long_running_recognize(config, audio)

# Block until operation complete
operation.wait_until_done!

raise operation.results.message if operation.error?

results = operation.response.results

#... <code to process results> ...
```

----
----

Our API reference docs demonstrate various ways to deal with LROs, which is pretty bomb.
https://googleapis.github.io/google-cloud-ruby/docs/google-cloud-speech/latest/Google/Cloud/Speech/V1/SpeechClient.html#long_running_recognize-instance_method

```ruby
# Register a callback during the method call.
operation = speech_client.long_running_recognize(config, audio) do |op|
  raise op.results.message if op.error?
  op_results = op.results
  # Process the results.

  metadata = op.metadata
  # Process the metadata.
end

# Or use the return value to register a callback.
operation.on_done do |op|
  raise op.results.message if op.error?
  op_results = op.results
  # Process the results.

  metadata = op.metadata
  # Process the metadata.
end

# Manually reload the operation.
operation.reload!

# Or block until the operation completes, triggering callbacks on
# completion.
operation.wait_until_done!
```